sdk/identity/Azure.Identity/src/Credentials/DefaultAzureCredentialOptions.cs

```diff
@@ -11,11 +14,33 @@ namespace Azure.Identity
     /// </summary>
     public class DefaultAzureCredentialOptions : TokenCredentialOptions
     {
+        private string _tenantId = GetNonEmptyStringOrNull(EnvironmentVariables.TenantId);
+
+        /// <summary>
+        /// The tenant ID the credential will authenticate to by default. If not specified the credential will authenticate to any requested tenant, and will default to the tenant to which the chosen authetication method was originally authenticated.
+        /// </summary>
+        public string TenantId
+        {
+            get
+            {
+                return _tenantId;
+            }
+            set
+            {
+                _tenantId = value;
+                InteractiveBrowserTenantId = value;
+                SharedTokenCacheTenantId = value;
+                VisualStudioCodeTenantId = value;
+                VisualStudioTenantId = value;
+            }
```

Comment:
I think this behavior might be unexpected for developers. Should we consider throwing here if one of the individual TenantIds is set to a different value than the incoming `value`?

Reply:
It's a good point, throwing would probably be preferred to overwriting a previously set value. Do you think the error message should suggest changing the ordering, or just blankly state they can't be different values?

Reply:
Updated the behavior to throw if setting `TenantId` after setting one of the more specific properties, and vice-versa.